metricbeat/autodiscover/builder/hints/metrics.go

```diff
@@ -202,6 +203,26 @@ func (m *metricHints) getHostsWithPort(hints common.MapStr, port int) ([]string,
 	return result, true
 }
 
+func (m *metricHints) checkHostPort(h string, p int) bool {
```

Comment:
Not to be solved in this PR, just food for thought: There are a couple of cases that I wonder how we could cover:
* IPv6 addresses can contain a `:80:` part, should `:port:` be rejected? Though this may never be a problem. If the host and the port are fixed, there is little use for autodiscover.
* Hosts that are not urls, e.g. this postgresql DSN: `host=${data.host} port=5432`

Reply:
autodiscover mostly covers only hints so im not sure how a person can hint the exact ip for the running pod.